network/peer/tls_config_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package peer_test
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/network/peer"
+)
+
+func TestValidateRSACertificate(t *testing.T) {
+	for _, testCase := range []struct {
+		description string
+		input       func() tls.ConnectionState
+		expectedErr error
+	}{
+		{
+			description: "Valid TLS cert",
+			input: func() tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
```

Comment:
I think these functions are referencing the wrong `t`. These would call operations on the `TestValidateRSACertificate` `t` rather than the inner `t.Run(...)` `t`.

Reply:
yes you're right. I moved them because of code review comments and didn't notice it.